### PR TITLE
support UCR runtime

### PR DESCRIPTION
**Pull request by kvish**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java [hunk 1]
```diff
@@ -927,6 +930,20 @@ private void getContainerInfoBuilder(Offer offer, Request request, String slaveN
 
                 containerInfoBuilder.setDocker(dockerInfoBuilder);
                 break;
+            case MESOS:
```
Comment (vinodkone) [inline src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java]: Dont' we need to expose this option in the plugin configuration?

File: src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java [hunk 2]
```diff
@@ -1007,6 +1024,7 @@ String generateJenkinsCommand2Run(int jvmMem,String jvmArgString,String jnlpArgS
 
         } else {
             LOGGER.fine("About to use default shell ....");
+            commandBuilder.setShell(true);
```
Comment (vinodkone) [inline src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java]: What is this for?

**Comment by vinodkone** [comment]
See #340

File: src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java (hunk 2 above)
Comment (kvish) [reply src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java]: removed

File: src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly [hunk 3]
```diff
@@ -38,7 +40,6 @@
                 </f:repeatableProperty>
             </f:entry>
         </f:radioBlock>
-        <f:radioBlock id="networking.USER" name="networking" title="${%User}" value="USER" inline="true" checked="${instance.networking == null || instance.networking == 'USER'}" />
```
Comment (vinodkone) [inline src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly]: Why was this deleted? Backwards compat?

File: src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly [hunk 4]
```diff
@@ -1,10 +1,10 @@
-<?jelly escape-by-default='true'?>
+?jelly escape-by-default='true'?>
```
Comment (vinodkone) [inline src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly]: open "<" missing?

File: src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:c="/lib/credentials">
     <f:entry title="${%Container Type}" field="type">
-        <f:radioBlock name="type" title="${%Docker}" value="DOCKER" inline="true" checked="true">
-            <f:entry title="${%Docker Image}" field="dockerImage" description="If using Docker, specify the docker image.">
+            <f:entry title="${%Docker Image}" field="dockerImage" description="Specify the docker image.">
```
Comment (vinodkone) [inline src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly]: Is this change backwards compatible? If someone upgrades the plugin and used previous config, sounds like it would fail to start? I think we probably want to just add another entry for docker image name inside the new radioblock for compat.

Comment (vinodkone) [reply src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly]: As discussed offline, we need to fix this for backwards compat.

Comment (kvish) [reply src/main/resources/org/jenkinsci/plugins/mesos/MesosSlaveInfo/ContainerInfo/config.jelly]: tested this and we should be good
- downgrades are not supported in jenkins so we are only concerned about uopgrade path
- When upgrading the containerType will be set to Docker correctly on the agent spec and the user can override this by selecting UCR
